src/ui/navigation/AppNavigator.tsx

```diff
@@ -292,8 +291,10 @@ function AppNavigator() {
 
   return (
     <AppStack.Navigator screenOptions={overlayScreenOptions}>
-      {pushAuthorizationStatus && pushAuthorizationStatus === messaging.AuthorizationStatus.NOT_DETERMINED ? (
-        <AppStack.Screen name={routeKeys.permissionGrantingPromptScreen} component={PermissionGrantingPrompt} />
+      {isPnPromptNeeded ? (
+        <AppStack.Screen name={routeKeys.permissionGrantingPromptScreen}>
+          {() => <PermissionGrantingPrompt skipPnPermission={skipPnPermission} />}
```

Comment:
Since this approach removes optimizations from react navigation to avoid re-renders you may want to use `React.memo` in `PermissionGrantingPrompt` or just call `usePushAuthorizationStatus` to get `skipPnPermission` instead of passing it as a prop

Reply:
Here it is removing the optimisations because of the render callback, that's true. But if the `usePushAuthorizationStatus` hook is called again it does not behaves as expected. Because it is a different instance of the hook.

Technically this component would be rendered very few times in reality. So for now it should be ok with `React.memo` which I have just added. Ideally we would need a context here but for this use case I think it is good enough.